test/functional/test_framework/util.py

```diff
@@ -326,6 +326,13 @@ def initialize_datadir(dirname, n, chain):
         os.makedirs(os.path.join(datadir, 'stdout'), exist_ok=True)
     return datadir
 
+def adjust_bitcoin_conf_for_pre_17(conf_file):
+    with open(conf_file,'r', encoding='utf8') as conf:
+        conf_data = conf.read()
+    with open(conf_file, 'w+', encoding='utf8') as conf:
+        conf_data_changed = re.sub(r'\[regtest\]', ' ', conf_data)
```

Comment:
```suggestion
        conf_data_changed = conf_data.replace('[regtest]', '')
```

Should be doable without relying on the regex module